src/cpp/shared/combat/army.cpp

```diff
@@ -314,25 +315,60 @@ void army::DoAttack(int isRetaliation) {
       }
     }
 
-    if(gIronfistExtra.combat.stack.abilityNowAnimating[primaryTarget]) {
-        if(primaryTarget->creatureIdx == CREATURE_CYBER_SHADOW_ASSASSIN) { // astral dodge animations
-            int dodgeAnimLen = 7;
-            primaryTarget->frameInfo.animationLengths[ANIMATION_TYPE_WINCE] = dodgeAnimLen;
-            for (int p = 0; p < dodgeAnimLen; p++) {
-              primaryTarget->frameInfo.animationFrameToImgIdx[ANIMATION_TYPE_WINCE][p] = 34 + p;
-            }
-
-            this->PowEffect(-1, 0, -1, -1);
-
-            // revert to usual animations after the first received attack
-            int winceAnimLen = 1;
-            primaryTarget->frameInfo.animationLengths[ANIMATION_TYPE_WINCE] = winceAnimLen;
-            primaryTarget->frameInfo.animationFrameToImgIdx[ANIMATION_TYPE_WINCE][0] = 50;            
-            gIronfistExtra.combat.stack.abilityNowAnimating[primaryTarget] = false;
+    if(primaryTarget->creatureIdx == CREATURE_CYBER_SHADOW_ASSASSIN) { // astral dodge animations
+      if(gIronfistExtra.combat.stack.abilityNowAnimating[primaryTarget]) {
+        int dodgeAnimLen = 7;
+        primaryTarget->frameInfo.animationLengths[ANIMATION_TYPE_WINCE] = dodgeAnimLen;
+        for (int p = 0; p < dodgeAnimLen; p++) {
+          primaryTarget->frameInfo.animationFrameToImgIdx[ANIMATION_TYPE_WINCE][p] = 34 + p;
         }
-    } else {
-      this->PowEffect(-1, 0, -1, -1);
+        gIronfistExtra.combat.stack.abilityNowAnimating[primaryTarget] = false;
+      } else {
+        // revert to usual animations after the first received attack
+        int winceAnimLen = 1;
+        primaryTarget->frameInfo.animationLengths[ANIMATION_TYPE_WINCE] = winceAnimLen;
+        primaryTarget->frameInfo.animationFrameToImgIdx[ANIMATION_TYPE_WINCE][0] = 50;
+      }
+    }
+
+    if(this->creatureIdx == CREATURE_CYBER_PLASMA_BERSERKER) {
+      if(gIronfistExtra.combat.stack.abilityNowAnimating[this]) {
+        this->frameInfo.animationLengths[ANIMATION_TYPE_MELEE_ATTACK_UPWARDS] = 
+        this->frameInfo.animationLengths[ANIMATION_TYPE_MELEE_ATTACK_FORWARDS] =
+        this->frameInfo.animationLengths[ANIMATION_TYPE_MELEE_ATTACK_DOWNWARDS] = 1;
+        this->frameInfo.animationLengths[ANIMATION_TYPE_MELEE_ATTACK_UPWARDS_RETURN] = 
+        this->frameInfo.animationLengths[ANIMATION_TYPE_MELEE_ATTACK_FORWARDS_RETURN] = 
```

Comment:
Please move all this into a data structure. Maybe one array for each new attack animation? Or maybe you can have a separate .bin file for the alternate animations, and you copy from there? Just.....make this idea take fewer than 40 lines to read.

Reply:
I re-emphasize this comment.